cmd/kubeapps-apis/plugin_test/mock_plugin.go

```diff
@@ -37,10 +39,21 @@ func (s TestPackagingPluginServer) GetAvailablePackageSummaries(ctx context.Cont
 	if s.Status != codes.OK {
 		return nil, status.Errorf(s.Status, "Non-OK response")
 	}
+	itemOffset, err := paginate.ItemOffsetFromPageToken(request.PaginationOptions.GetPageToken())
+	if err != nil {
+		return nil, err
+	}
+	summaries := s.AvailablePackageSummaries[itemOffset:]
+	pageSize := int(request.PaginationOptions.GetPageSize())
+	nextPageToken := ""
+	if pageSize > 0 && pageSize < len(summaries) {
+		summaries = summaries[:pageSize]
+		nextPageToken = fmt.Sprintf("%d", itemOffset+pageSize)
+	}
```

Comment:
This just updates the mock plugin to be a little more realistic for the `GetAvailablePackageSummaries` call, returning the correct (calculated) summaries based on the pagination options, and a calculated `nextPageToken` (rather than the stock one on the struct, which we can remove once we update the installed package summaries call as well).